packages/react/src/index.tsx

```diff
@@ -146,26 +164,33 @@ export function useWithContext(context: Context, parent = useConfidence()): Conf
     [parent, Value.serialize(context)],
   );
 
-  const [, setState] = useState(() => child.contextState);
-  useEffect(
-    () =>
-      child.subscribe(state => {
-        if (state === 'READY' || state === 'ERROR') setState(child.contextState);
-      }),
-    [child, setState],
-  );
   return child;
 }
 
+// this would be better named useContext, but would then collide with React.useContext
+// eslint-disable-next-line react-hooks/rules-of-hooks
+export function useConfidenceContext(confidence = useConfidence()): Context {
+  const [, setState] = useState(confidence.contextState);
+  useEffect(() => {
+    return confidence.delegate.contextChanges(() => setState(confidence.contextState));
+  });
+  return confidence.delegate.getContext();
+}
+
 export function useEvaluateFlag<T extends Value>(
   path: string,
   defaultValue: T,
   // eslint-disable-next-line react-hooks/rules-of-hooks
   confidence = useConfidence(),
 ): FlagEvaluation<Value.Widen<T>> {
   const evaluation = confidence.delegate.evaluateFlag(path, defaultValue);
-  // TODO make it a setting to _enable skip throwing_ on stale value.
-  if (evaluation.reason === 'ERROR' && evaluation.errorCode === 'NOT_READY' && 'then' in evaluation) throw evaluation;
+  const [, setState] = useState(() => confidence.contextState);
+  useEffect(() => {
+    return confidence.delegate.contextChanges(() => {
+      setState(confidence.contextState);
+    });
+  });
+  if ('then' in evaluation) throw evaluation;
```

Comment:
We might have a slight problem here. Since we don't kick off a resolution until React calls the "then" method, we risk a "waterfall" loading behaviour. I.e. there are two sibling components that both are stale and need to fetch. The first component renders and throws a suspense. Then the second component wont be rendered until the first promise resolves and at that point the second component will throw and initiate a fetch... Actually, this shouldn't be the case in React 18 (where the second component will render to cache a potential suspense), but there was just rather big controversy about React 19 changing this behaviour, but then last minute they delayed React 19 due to backlash from the community. (See https://tkdodo.eu/blog/react-19-and-suspense-a-drama-in-3-acts) All that said, I think we'd be better off keeping track of which ConfidenceReact instances are currently serving flags to the UI and proactively initiate fetches for them when the context changes. Will have a quick look at this!

Reply:
Fixed.